File: Booking-Service/src/main/java/com/theater/booking/entity/Booking.java
package com.theater.booking.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "booking")
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Booking {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "user_mail", nullable = false)
    private String userMail;

    @Column(name = "show_id", nullable = false)
    private Long showId;

    @Column(name = "booking_date", nullable = false)
    private LocalDateTime bookingDate;

    @Column(name = "show_time", nullable = false)
    private LocalDateTime showTime;

    @Column(name = "seats")
    private String seatId;

    @Column(name = "booking_reference", nullable = false, unique = true)
    private String bookingReference;

    @Column(name = "status", nullable = false)
    private String status;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;
   
}

File: Frontend/frontend/src/app/models/Booking.ts
export class Booking {
  id: number;
  userMail: string;
  showId: number;
  bookingDate: Date;
  totalPrice: number;
  seatId: number;
  status: string;
  createdAt: Date;
  updatedAt: Date;
  showTime: Date;
  userName: string;

  constructor(
    id: number,
    userMail: string,
    showId: number,
    bookingDate: Date,

    totalPrice: number,
    seatId: number,
    status: string,
    createdAt: Date,
    updatedAt: Date,
    showTime: Date,
    userName: string
  ) {
    this.id = id;
    this.userMail = userMail;
    this.showId = showId;
    this.bookingDate = bookingDate;
    this.totalPrice = totalPrice;
    this.seatId = seatId;
    this.status = status;
    this.createdAt = createdAt;
    this.updatedAt = updatedAt;
    this.showTime = showTime;
    this.userName = userName;
  }
}
